### PR TITLE
Add lightweight serving and support tgi parameter

### PR DESCRIPTION
**Pull request by hzjane**
### 4. How to test?
- [ ] N/A
- [ ] Unit test: Please manually trigger the PR Validation [here](https://github.com/intel-analytics/ipex-llm-workflow/actions/workflows/llm-PR-validation.yml) by inputting the PR number (e.g., `1234`). And paste your action link here once it has been successfully finished.
- [ ] Application test
- [ ] Document test
- [ ] ...

### 5. New dependencies

- [ ] New Python dependencies
       - Dependency1 
       - Dependency2
       - ...
- [ ] New Java/Scala dependencies and their license
       - Dependency1 and license1
       - Dependency2 and license2
       - ...

### CODE REVIEW

File: python/llm/src/ipex_llm/transformers/pipeline_parallel.py [hunk 1]
```diff
@@ -697,14 +697,14 @@ async def add_request(self, tokenizer):
...
-        plain_texts = [req.prompt for req in prompt_requests]
+        plain_texts = [req.inputs for req in prompt_requests]
         inputs = tokenizer(plain_texts, return_tensors="pt", padding=True)
         input_ids = inputs.input_ids.to(f'xpu:{self.rank}')
         attention_mask = inputs.attention_mask.to(f'xpu:{self.rank}')
         new_batch = BatchTask(
             batch_id="batch_" + str(uuid.uuid4()),
             request_ids=request_ids,
-            max_tokens=max([req.n_predict for req in prompt_requests]),
+            max_tokens=max([req.parameters.mex_new_tokens for req in prompt_requests]),
```
Comment (xiangyuT) [inline python/llm/src/ipex_llm/transformers/pipeline_parallel.py:707]: mex_new_tokens -> max_new_tokens

File: python/llm/src/ipex_llm/serving/fastapi/api_server.py [hunk 2]
```diff
@@ -280,18 +291,19 @@ async def create_chat_completion(request: ChatCompletionRequest):
...
         n_predict = request.max_tokens
-    prompt_request = PromptRequest(
-        prompt=request.prompt,
-        n_predict=n_predict,
-        req_type="completion"
+    inputs_request = InputsRequest(
+        inputs=request.prompt,
+        parameters=Parameters(max_new_tokens=n_predict),
+        stream=request.stream,
+        req_type="chat"
```
Comment (xiangyuT) [inline python/llm/src/ipex_llm/serving/fastapi/api_server.py:301]: chat -> completion?

File: python/llm/src/ipex_llm/serving/fastapi/api_server.py [hunk 3]
```diff
@@ -25,12 +25,20 @@
...
 
 
+class InputsRequest(BaseModel):
+    inputs: str
+    parameters: Optional[Parameters] = None
+    stream: Optional[bool] = False
+    req_type: str = 'completion'
+
+
 class PromptRequest(BaseModel):
```
Comment (xiangyuT) [inline python/llm/src/ipex_llm/serving/fastapi/api_server.py:42]: Maybe merge `PromptRequest` to `InputsRequest`?

Comment (hzjane) [reply python/llm/src/ipex_llm/serving/fastapi/api_server.py:42]: I will remove PromptRequest later

File: docker/llm/serving/xpu/docker/Dockerfile [hunk 4]
```diff
@@ -33,6 +33,7 @@ COPY ./start-vllm-service.sh           /llm/
 COPY ./benchmark_vllm_throughput.py   /llm/
 COPY ./start-fastchat-service.sh       /llm/
 COPY ./start-pp_serving-service.sh       /llm/
+COPY ./start-lightweight_serving-service.sh       /llm/
```
Comment (xiangyuT) [inline docker/llm/serving/xpu/docker/Dockerfile:36]: Should we check the consistent use of `_` and `-`？

File: python/llm/example/GPU/Lightweight-Serving/README.md [hunk 5]
```diff
@@ -0,0 +1,208 @@
...
+    "temperature": 1.0,
+    "do_sample": false,
+    "top_k": 5,
+    "tok_p": 1.0
+  },
+  "stream": false
+}' http://localhost:8000/generate
+```
+
+#### generate_stream()
```
Comment (glorysdj) [inline python/llm/example/GPU/Lightweight-Serving/README.md:148]: change to a url like /v1/chat/completions?

File: python/llm/example/GPU/Lightweight-Serving/README.md [hunk 6]
```diff
@@ -0,0 +1,208 @@
...
+- `--repo-id-or-model-path REPO_ID_OR_MODEL_PATH`: argument defining the huggingface repo id for the model (e.g. `meta-llama/Llama-2-7b-chat-hf` and `meta-llama/Llama-2-13b-chat-hf`) to be downloaded, or the path to the huggingface checkpoint folder. It is default to be `'meta-llama/Llama-2-7b-chat-hf'`.
+- `--low-bit LOW_BIT`: Sets the low bit optimizations (such as 'sym_int4', 'fp16', 'fp8' and 'fp6') for the model. It is default to be `sym_int4`.
+- `--port PORT`: The serving access port. It is default to be `8000`.
+
+
+### 5. Sample Input and Output
+
+We can use `curl` to test serving api
+
+#### generate()
```
Comment (glorysdj) [inline python/llm/example/GPU/Lightweight-Serving/README.md:127]: change to a url like /v1/chat/completions?

**Review by glorysdj — APPROVED** [review]
LGTM

**Pull request merged**